index_test.go

```diff
@@ -128,20 +131,39 @@ func TestBasic(t *testing.T) {
 			// --------------0123456789012345678901234567890123
 		})
 
-	res := searchForTest(t, b, &query.Substring{
-		Pattern:       "water",
-		CaseSensitive: true,
+	t.Run("LineMatch", func(t *testing.T) {
```

Comment:
In this file, I split each relevant test into two sub-tests: `LineMatch` and `ChunkMatch`, one which runs with with `ChunkMatches: true`, and one that maintains the old behavior of the tests. This provides a decent amount of confidence in the correctness of the `ChunkMatch` implementation, and also will make it easy to rip out all the line match stuff in the case we want to complete the migration to chunk matches.